### ISSUE TITLE
highlights.scm can't be rewrite

### ISSUE THREAD

**Issue by kevinhwang91** [nvim-treesitter/nvim-treesitter#1710]
**Describe the bug**

https://github.com/nvim-treesitter/nvim-treesitter/pull/1690
I don't like 
```
((identifier) @constant
 (#match? @constant "^[A-Z][A-Z_0-9]*$"))
```
and I want to delete the highlighting behavior by rewriting my `~/.config/nvim/queries/lua/highlights.scm`. But it doesn't work.

**To Reproduce**
1. only delete the above snippet code and write the content to `~/.config/nvim/queries/lua/highlights.scm`.

**Expected behavior**
Don't highlight the upper chars.

**Comment by vigoux** [comment]
You should be able to overwrite queries using `:h set_query()`

**Comment by kevinhwang91** [comment]
> - if the file is in any `after/queries/` folder, then it will be used to extend the already defined
  queries.
> - Otherwise, it will be used as a base to define the query, the first query found (with the highest
  priority) will be the only one to be used.

Did I misunderstand the content from README?

**Comment by vigoux** [comment]
that's the remainder of an old way of doing it, we really should remove it.

**Comment by kevinhwang91** [comment]
That's saying I must program for the rewriting?

**Comment by vigoux** [comment]
You will have to program some lua yes, that's exactly what I am saying

**Comment by kevinhwang91** [comment]
IMO. It's so weird for customizing highlighting style by programing with hard code. I prefer the old way by creating a scm file to rewrite the style. Did old way has a technical issue?

**Comment by vigoux** [comment]
the old way was actually pretty unclear about how it worked, and not really stable. We decided to remove it in the end.

**Comment by vigoux** [comment]
What we could (and should to tbh) is provide a directive in the scm files to override all the previously found files

**Comment by theHamsta** [comment]
Why not use the PR with the ignore files? Then, you wouldn't have to rely on the ordering and overload the query DSL with more nvim-stuff

**Comment by vigoux** [comment]
Because that requires to write lua, and some people may not like it.

We need to find a lips-only way of overriding things

**Comment by akinsho** [comment]
Does this apply for all custom queries? does the `queries` directory in the runtimepath do anything e.g. if I wanted to add injections for a language could I still add my own `.scm` file somewhere or does any attempt at add or overriding a query now require adding it via set query?

**Comment by vigoux** [comment]
Any _overwritting_ of queries requires `set_query`.
Any _addition_ of queries can be done through `.scm` files.

**Comment by theHamsta** [comment]
@vigoux probably SCM files should behave like traditional runtime files (with the overwrite behavior). The new concat-behavior can be explicitly be requested by the modelines

**Comment by vigoux** [comment]
It's pretty difficult to see if its better or not, I'd open an issue [upstream](https://github.com/neovim/neovim) and discuss there tbh/

**Comment by gpanders** [comment]
I just got bit by this too. It used to be that I could override specific queries provided by nvim-treesitter by creating a file in `after/queries/{lang}/highlights.scm` with my own queries to tweak the highlights, but it looks like this no longer works. I found that method quite convenient and I'm surprised that it was removed. I'm definitely in favor of supporting some kind of method to override queries by just using a `.scm` file.
